### DISCUSSION TITLE
What can I do to resolve these errors?

### DISCUSSION THREAD

**Discussion by Ifriqiya** [larastan/larastan#1469] [Q&A] [answered]
I have the following errors that I need help with:

1. This portion of the code in Handler.php
```php
    /**
     * Prepare exception for rendering.
     *
     * @param  \Throwable  $e
     * @return mixed
     */
    public function render($request, Throwable $e)
    {
        $response = parent::render($request, $e);

        if ($response->status() === 419) {
            return back()->with([
                'message' => 'The page expired, please refresh and try again.',
            ]);
        }
      //
}
``` 

throws this error `Call to an undefined method Symfony\Component\HttpFoundation\Response::status()`

2. This section of code in a controller that uses the session helper function 
```php
   public function payPendingBill(Request $request)
    {
       //
        $pending_bill_id = $request->session()->pending_bill_id;
      //
    }
```
throws this error `Access to an undefined property Illuminate\Contracts\Session\Session::$pending_bill_id`

3. This section of code that dispatches a job 
```php
      //
      if ($paid_bill->bill_type === 'Performance Fee') {
            return SuccessfulPerformanceFeePayment::dispatch($this->user, $paid_bill);
        } else {
            return SuccessfulSetupFeePayment::dispatch(Auth::user(), $paid_bill);
        }
        //
```
throws this error ` Result of static method App\Events\SuccessfulPerformanceFeePayment::dispatch() (void) is     
         used. And Result of static method App\Events\SuccessfulSetupFeePayment::dispatch() (void) is used`

4. This code 
```php
     //
     return request()->session()->put('pending_bill_id', $pending_bill->id);
``` 
throws this error `Result of method Illuminate\Contracts\Session\Session::put() (void) is used.`

5. This code segment 
```php
      // 
        $recipients = $messages->map->recipient;
        $senders = $messages->map->user;
        $allparticipants = $recipients->merge($senders);

        $participants = $allparticipants->whereNotIn('id', Auth::id());
        //
```
throws this error `Parameter #2 $values of method Illuminate\Support\Collection<int,App\Models\User|null>::whereNotIn()  
         expects Illuminate\Contracts\Support\Arrayable|iterable, int|string|null given.`

6. This code segment
```php
     //
    public function read(User $user)
    {
        $user = Auth::user();
        $user->unreadNotifications->markAsRead();
    }
    //
```
throws this error `Access to an undefined property App\Models\User::$unreadNotifications.`

7. This code segment 
```php
     /**
     * Handle the event.
     *
     * @param  Lockout  $event
     * @return void
     */
    public function handle(Lockout $event)
    {
        $admin = env('MAIL_FROM_ADDRESS');
        $recipients = [$admin, $event->user];
        Notification::send($recipients, new LockoutNotice);
    }
```
throws this error `Access to an undefined property Illuminate\Auth\Events\Lockout::$user.`

8. This code segment
```php
     /**
     * Handle the event.
     *
     * @param  Registered  $event
     * @return void
     */
    public function handle(Registered $event)
    {
        $admin = env('MAIL_FROM_ADDRESS');
        Notification::route('mail', [
            $admin => 'SmartWealth',
        ])
        ->notify(new NewlyRegisteredUser($event->user));
    }
```
throws this error `Parameter #1 $user of class App\Notifications\NewlyRegisteredUser constructor expects App\Models\User,
         Illuminate\Contracts\Auth\Authenticatable given`

Thanks.

**Comment by canvural** [comment]
Please format the code better. It's hard to read.

**Reply by szepeviktor** [reply]
&#96;&#96;&#96;php
code ....
&#96;&#96;&#96;

**Reply by Ifriqiya** [reply]
> Please format the code better. It's hard to read.

Hi @canvural, so sorry about that. Is this OK now?

**Comment by szepeviktor** [comment]
When a function returns `void`
```php
$do->this();
return;
```

**Reply by Ifriqiya** [reply]
> When a function returns `void`
> 
> ```
> $do->this();
> return;
> ```

Thanks @szepeviktor, this resolved questions 3 & 4.

**Comment by marineusde** [comment, marked as answer]
Number 1:
```php
$response->getStatusCode()
```

Number 2:
```
$pending_bill_id = $request->session()->get('pending_bill_id');
```

Number 5:
```php
$participants = $allparticipants->where('id', '!=', Auth::id());
```
Cause whereIn() needs an array, not just an integer.

Number 6:
Are you sure, there is a method unreadNotifications in User-Model? Please post it.

Number 7:
There is no User-Object in the Lockout-Class. Try to get the real class name of $event

Number 8:
Try this:
```
public function handle(Registered $event)
{
    $admin = env('MAIL_FROM_ADDRESS');
    /** @var User $user */
    $user = $event->user;
    Notification::route('mail', [
        $admin => 'SmartWealth',
    ])
    ->notify(new NewlyRegisteredUser($user));
}
```

**Reply by Ifriqiya** [reply]
Hello @zwhhz, thanks for this. The unreadNotifications method is a laravel method that's available on the default user model via eloquent notifications relationship. See https://laravel.com/docs/9.x/notifications#accessing-the-notifications

**Reply by marineusde** [reply]
Ah, now I found it.

First Idea:     
```
/** @var User $user */
$user = Auth::user();

/** @var DatabaseNotificationCollection $notifications */
$notifications = $user->unreadNotifications()->get();

$notifications->markAsRead();
```

Second Idea:

In User-Model:
```
/**
 * @property DatabaseNotificationCollection $unreadNotifications
 */
class User extends Authenticatable
```

In your Class:
```
/** @var User $user */
$user = Auth::user();

$user->unreadNotifications->markAsRead();
```

**Comment by Ifriqiya** [comment]
Thanks for this. I couldn't get either to work but I had long used this: ` $user->unreadNotifications()->update(['read_at' => now()]);` which removed the error message.
